src/components/DataSlice.js

```diff
@@ -0,0 +1,14 @@
+const initialState = {
+  data: [],
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'SET_DATA':
+      return { ...state, data: action.payload };
+    default:
+      return state;
+  }
+};
+
```

Comment:
[OPTIONAL] This is an old way of doing things and i strongly recommend using the createslisce method shipped together by redux toolkit to make your code syntax more modern

Reply:
I will implement in the next project